transports/websocket/src/desktop.rs

```diff
@@ -236,25 +236,7 @@ where
     }
 
     fn nat_traversal(&self, server: &Multiaddr, observed: &Multiaddr) -> Option<Multiaddr> {
-        let mut server = server.clone();
-        let last_proto = match server.pop() {
-            Some(v @ Protocol::Ws) | Some(v @ Protocol::Wss) => v,
-            _ => return None,
-        };
-
-        let mut observed = observed.clone();
-        match observed.pop() {
-            Some(Protocol::Ws) => false,
-            Some(Protocol::Wss) => true,
-            _ => return None,
-        };
-
-        self.transport
-            .nat_traversal(&server, &observed)
-            .map(move |mut result| {
-                result.append(last_proto);
-                result
-            })
+        self.transport.nat_traversal(server, observed)
```

Comment:
In accordance to my other comment, this should remove the `/ws` or `/wss` suffix from `server`, or return `None` if there's none.

Reply:
Doesn't it make more sense that each `Transport` assumes `server` matches an address returned by `listen_on` before, so that there's no overlap between implementations?

Reply:
I find the prefix translation quite intuitive. Could you maybe explain a bit where a possible overlap would become problematic?

Reply:
I haven't really though about it, but there may be issues with the relay protocol.

Reply:
Our relay transport only supports dialing IIRC.